src/transformStateWithClones.js

```diff
@@ -1,13 +1,39 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object}  currentState
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+      states.push({ ...currentState });
+    }
+
+    if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+      states.push({ ...currentState });
+    }
+
+    if (action.type === 'removeProperties') {
```

Comment:
The current implementation repeats the action of pushing the current state to the states array in each if block. According to the checklist, you should perform this action just once after processing each action.